opta/meister.py

```diff
@@ -0,0 +1,172 @@
+import atexit
+import inspect
+import os
+import sys
+import time
+from asyncio import Lock, iscoroutinefunction
+from functools import wraps
+from math import ceil, floor, log10
+from typing import Any, Callable, Dict, List, Tuple, TypedDict
+from uuid import UUID, uuid4
+
+from colored import attr, fg
+
+
+def round_sig(x: float, sig: int = 3) -> float:
+    return round(x, sig - int(floor(log10(abs(x)))) - 1)
+
+
+class TimedCall(TypedDict):
+    id: UUID
+    function_name: str
+    positional_args: Tuple[Any, ...]
+    keyword_args: Dict[str, Any]
+    start_time: float
+    end_time: float
+    caller_file: str
+    caller_line: int
+
+
+MEISTER_RECORDS: List[TimedCall] = []
+MEISTER_ASYNC_LOCK = Lock()
+MEISTER_REPORT_SIZE = 20
+MEISTER_TIME_THRESHOLD = 0.1
+MEISTER_REPORT_ENABLED = True
+BLOCK_SYMBOL = "\u2588"
+MEISTER_START_TIME = time.time()
```

Comment:
```suggestion
ASYNC_LOCK: Final = Lock()
REPORT_SIZE: Final = 20
TIME_THRESHOLD: Final = 0.1
REPORT_ENABLED: bool = True
BLOCK_SYMBOL: Final = "\u2588"
START_TIME: Final = time.time()
```

- We already know these are constants/variables for meister, since we are in that module. No need to duplicate the module name in the variable names.
- Added `typing.Final` to some variables to hint that they are constant.